spec/requests/registrations/user_profile_spec.rb

```diff
@@ -23,39 +23,104 @@
   end
 
   describe "POST /registrations/user-profile" do
-    it "creates an induction coordinator profile" do
-      post "/registrations/user-profile", params: { user: {
-        full_name: full_name,
-        email: email,
-      } }
-
-      expect(created_user.full_name).to eq(full_name)
-      expect(response).to redirect_to(:registrations_verification_sent)
-      expect(created_user.induction_coordinator_profile.schools).to include(school)
+    before do
```

Comment:
So did we just decide to go with the tried and tested mocks / stubs in the end. I remember there were some complex alternatives in the discussions earlier. Probably best to keep it simple though. I don't think there was a strong case for the complex solution.

Reply:
Well, there is a second part in cypress tests coming up. They do look more 'real life'. For the purposes of unit tests, I don't believe there is much point in having anything more complicated than a stub checking what method was called with what arguments on mailers.

In cypress tests we play with `ActionMailer::Base.deliveries` which look a lot more like real requests to notify :)

Reply:
One of the alternatives was along the lines of building something that could mimic notify I think. That's what I was referring to.

Reply:
When working on cypress tests I really warmed up to the default Railsy idea of 'when tests are running, put the emails in an array rather than sending them'. That imo leads to a clean and pain-free way of verifying our emails, without the need to include a notify stub. 

And well, for unit test purposes, I really think stubbing the mailer is the way to go - there is barely any logic in the mailer itself, and if there is we can unit test that specifically at a lower level. Stubbing notify project-wide in tests doesn't seem to provide much benefit.